lib/api.js

```diff
@@ -38,93 +35,28 @@ function worker(renditionCallback, options = {}) {
     }
 
     return actionWrapper(async function (params) {
+        /*
+        // Debug: Use sleep to ensure tests have time to grab logs
         const { promisify } = require('util');
         const sleep = promisify(setTimeout);
         await sleep(2000);
         console.log("## worker(renditionCallback) -> actionWrapper");
+        //*/
 
-        // TODO:
-        // Move all of this pipeline logic into a class `AssetComputePipeline`
-        // this class will work similarly to `AssetComputeWorker` and catch errors in the pipeline and send events/metrics
-        // look at error type
-
-        // TODO:
-        // try/catch around engine to catch system errors
-        // and throw a GenericError if it is not an error type we recognize
-        const pipeline = new Engine(params);
-        const transformers = await buildTransformers(renditionCallback, options);
-        transformers.forEach( eachTransformer => {
-            console.log('##### transformer', eachTransformer.name);
-            pipeline.registerTransformer(eachTransformer);
+        // if any rendition needs pipeline, all will go through pipeline
+        const usePipeline = params.renditions.some(rendition => {
+            return rendition.pipeline === true;
```

Comment:
IIUC, in mvp, pipeline will only
- support 1 rendition per instance of pipeline. 
- only send sensei request to pipeline. Other renditions would continue to go to sdk.
So don't we need to split the renditions into pipeline vs sdk ?

Reply:
> support 1 rendition per instance of pipeline.
Since we are splitting sensei requests into separate renditions from aem, we do need to support multiple pipeline renditions per container for MVP.  There are two ways to solve this:
- SDK runs multiple pipelines in parallel for each rendition
- add support for multiple renditions (run multiple plans) in the pipeline
There is a ticket in the next sprint to decide how we will do this: https://jira.corp.adobe.com/browse/NUI-1284

> So don't we need to split the renditions into pipeline vs sdk ?

we do, and we actually have a design ticket next sprint to discuss this: https://jira.corp.adobe.com/browse/NUI-1338
(currently we were thinking core would do this dynamic logic, but it could move to the SDK. there are pro's and con's and each)

Reply:
> only send sensei request to pipeline. Other renditions would continue to go to sdk.

That is correct

Reply:
`only send sensei request to pipeline. Other renditions would continue to go to sdk.` => Indeed. We would not even need the pipeline flag currently/technically, but it is explicitly requested in the story. 

Also I may have ideas to deal with multiple renditions, but I didn't implement anything because, as Jamie noted, the design ticket is upcoming (and I don't want to put a design we didn't discuss in the mob branch just now).